src/runtime/plugin.ts

```diff
@@ -31,7 +33,27 @@ export default defineNuxtPlugin(async (nuxtApp) => {
 
   // Only fetch session if it was not yet initialized server-side
   if (typeof data.value === 'undefined' && !nitroPrerender && !disableServerSideAuth) {
-    await getSession()
+    // Restore the session data from the cookie if it exists
```

Comment:
Changing this also results in changes to how the `authjs` provider works, which is causing the tests to fail. Please look into ensuring that your new logic does not effect this provider 😊

Reply:
fixed authjs failing problem. Sorry for ugly commit history, I'm not too familiar with git.